src/python/pants/backend/python/tasks/build_local_python_distributions.py

```diff
@@ -39,52 +43,69 @@ def product_types(cls):
   def prepare(cls, options, round_manager):
     round_manager.require_data(PythonInterpreter)
 
+  @classmethod
+  def implementation_version(cls):
+    return super(BuildLocalPythonDistributions, cls).implementation_version() + [('BuildLocalPythonDistributions', 1)]
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(BuildLocalPythonDistributions, cls).subsystem_dependencies() + (LLVM.scoped(cls),)
+
+  @memoized_property
+  def llvm_base_dir(self):
+    return LLVM.scoped_instance(self).select()
+
   @property
   def cache_target_dirs(self):
     return True
 
   def execute(self):
     dist_targets = self.context.targets(is_local_python_dist)
-    build_graph = self.context.build_graph
 
     if dist_targets:
       with self.invalidated(dist_targets,
                             fingerprint_strategy=DefaultFingerprintStrategy(),
                             invalidate_dependents=True) as invalidation_check:
+        interpreter = self.context.products.get_data(PythonInterpreter)
+        sandboxed_interpreter = SandboxedInterpreter(self.llvm_base_dir, interpreter)
+
         for vt in invalidation_check.invalid_vts:
           if vt.target.dependencies:
             raise TargetDefinitionException(
               vt.target, 'The `dependencies` field is disallowed on `python_dist` targets. '
                          'List any 3rd party requirements in the install_requirements argument '
                          'of your setup function.'
             )
-          self._create_dist(vt.target, vt.results_dir)
+          self._create_dist(vt.target, vt.results_dir, sandboxed_interpreter)
 
         for vt in invalidation_check.all_vts:
           dist = self._get_whl_from_dir(os.path.join(vt.results_dir, 'dist'))
           req_lib_addr = Address.parse('{}__req_lib'.format(vt.target.address.spec))
           self._inject_synthetic_dist_requirements(dist, req_lib_addr)
           # Make any target that depends on the dist depend on the synthetic req_lib,
           # for downstream consumption.
-          for dependent in build_graph.dependents_of(vt.target.address):
-            build_graph.inject_dependency(dependent, req_lib_addr)
-
-  def _create_dist(self, dist_tgt, dist_target_dir):
-    """Create a .whl file for the specified python_distribution target."""
-    interpreter = self.context.products.get_data(PythonInterpreter)
+          for dependent in self.context.build_graph.dependents_of(vt.target.address):
```

Comment:
This is same with existing codes except with build_graph is not stored in a local variables, right?  Is there a reason for this change? I am afraid that I am missing something.

Reply:
Yes, it's the same. Since the build graph is just a property access making it into a separate variable was a bit confusing to me. It doesn't affect the logic.